File: 2962.count-subarrays-where-max-element-appears-at-least-k-times.py
#
# @lc app=leetcode id=2962 lang=python3
#
# [2962] Count Subarrays Where Max Element Appears at Least K Times
#

# @lc code=start
class Solution:
    def countSubarrays(self, nums: list[int], k: int) -> int:
        max_num = max(nums) 
        left, ans = 0, 0
        for i in range(len(nums)):
            if nums[i] == max_num:
                k -= 1
            while k == 0:
                if nums[left] == max_num:
                    k += 1
                left += 1
            ans += left
        return ans



          
        
# @lc code=end

File: 2962.count-subarrays-where-max-element-appears-at-least-k-times.java
/*
 * @lc app=leetcode id=2962 lang=java
 *
 * [2962] Count Subarrays Where Max Element Appears at Least K Times
 */

// @lc code=start
class Solution {
    public long countSubarrays(int[] nums, int k) {
        int mx = 0;
        for(int num: nums){
            mx = Math.max(mx, num);
        }

        long ans = 0;
        int left = 0;
        for(int right = 0; right < nums.length; right++){
            if(nums[right] == mx){
                k--;
            }
            while(k == 0){
                if(nums[left] == mx){
                    k++;
                }
                left++;
            }
            ans += left;
        }
        return ans;
    }
}
// @lc code=end

